### PR TITLE
ac-2: Added time feature in pre registration

### PR DESCRIPTION
**Pull request by Anunaya07**
## Proposed changes
Added time feature in pre registration

## Types of changes

_Put an `x` in the boxes that apply_

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Other (please describe):

## Checklist

_Put an `x` in the boxes that apply_

-   [x] My code follows the [style guidelines of this project](https://docs.google.com/document/d/1GI2Hile8UbGZ82gFe1y4wAVPcNPXip1cmXTzog1S41U/edit)
-   [x] I have performed a self-review of my own code
-   [ ] I have created new branch for this pull request
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [ ] My changes generate no new warnings
-   [ ] My changes does not break the current system and it passes all the current test cases.

### CODE REVIEW

**Comment by Aakash2611Wagh** [comment]
Approved. @akshatnema

**Comment by akshatnema** [comment]
Changes in the models.py so a review is required from @PranshuNayak and @samay-rgb.

**Comment by akshatnema** [comment]
@Anunaya07 Kindly push the migrations of this PR as there are changes in the database.

File: FusionIIIT/applications/academic_information/models.py [hunk 1]
```diff
@@ -274,6 +274,8 @@ class Calendar(models.Model):
     '''
     from_date = models.DateField()
     to_date = models.DateField()
+    start = models.TimeField(default='20:00')
+    end = models.TimeField(default='20:00')
```
Comment (PranshuNayak) [inline FusionIIIT/applications/academic_information/models.py:278]: rename each to start_time & end_time
refer [https://stackoverflow.com/questions/16027516/can-i-set-a-specific-default-time-for-a-django-datetime-field](url)
and modify default accordingly. Also push a single migration file after changes.

File: FusionIIIT/applications/academic_information/models.py [hunk 2]
```diff
@@ -274,6 +274,8 @@ class Calendar(models.Model):
     '''
     from_date = models.DateField()
     to_date = models.DateField()
+    start_time = models.TimeField(default='20:00')
+    end_time = models.TimeField(default='20:00')
     description = models.CharField(max_length=40)
```
Comment (PranshuNayak) [inline FusionIIIT/applications/academic_information/models.py:279]: Provide the default value like [this](https://stackoverflow.com/questions/16027516/can-i-set-a-specific-default-time-for-a-django-datetime-field)

**Review by PranshuNayak — APPROVED** [review]
@Anunaya07 push only a single migration file

**Review by samay-rgb — APPROVED** [review]

**Comment by PranshuNayak** [comment]
@akshatnema kindly revert this.
@Anunaya07 still there are 2 migrations files , kindly rectify

**Comment by Anunaya07** [comment]
@PranshuNayak @samay-rgb  I am getting this error if I delete initial_0001 and tried to re-create it

django.db.migrations.exceptions.NodeNotFoundError

**Comment by PranshuNayak** [comment]
> @PranshuNayak @samay-rgb I am getting this error if I delete initial_0001 and tried to re-create it
> 
> django.db.migrations.exceptions.NodeNotFoundError

Provide the full traceback error, the one you see on the terminal.

**Comment by Anunaya07** [comment]
Screenshot 2023-03-30 144408

**Comment by PranshuNayak** [comment]
Follow these steps:
1. Migrate back to 0001 in academic procedures
2. delete the other migrations
3. use the column names as suggested before and makemigrations
4. Push that file only.

**Comment by Anunaya07** [comment]
Can you also specify the commands?

**Comment by Anunaya07** [comment]
@PranshuNayak can you also specify the commands?

**Comment by PranshuNayak** [comment]
Refer [this](https://docs.djangoproject.com/en/3.2/topics/migrations/#reversing-migrations)

**Comment by PranshuNayak** [comment]
@Anunaya07 are u able to resolve the issue ?

**Comment by Anunaya07** [comment]
No

**Comment by PranshuNayak** [comment]
What steps have you done?

**Comment by Anunaya07** [comment]
@PranshuNayak I have done all that you suggested but still wasn't able to resolve it.

**Pull request merged**